tests/product/base_product_case.py

```diff
@@ -409,3 +428,12 @@ def escape_for_regex(self, expected):
         expected = expected.replace('(', '\(')
         expected = expected.replace('+', '\+')
         return expected
+
+
+def docker_only(original_function):
+    def inner(self, *args, **kwargs):
+        if type(getattr(self, 'cluster')) is DockerCluster:
+            original_function(self, *args, **kwargs)
+        else:
+            raise Exception('Docker only test, running on real cluster')
```

Comment:
I'm not sure we want to raise an exception for non-docker cluster.  Otherwise we won't be able to get a clean test run for non-docker clusters.  maybe just print a warning message?